### ISSUE TITLE
GroupCoordinatorNotAvailable error

### ISSUE THREAD

**Issue by erikvullings** [spreedly/kaffe#43]
I have been using Kaffe successfully for connecting to Kafka running Docker on Windows 10 (my Docker-compose file can be found [here](https://gist.github.com/erikvullings/b6363444fb232477ef381b1bb7dcbb32)). However, when updating the Kafka Docker images, it stopped working, and I received below exception:
```c
15:15:10.634 [info]  group coordinator (groupId=test-consumer-group,memberId=,generation=0,pid=#PID<0.179.0>):
failed to join group
reason::GroupCoordinatorNotAvailable
```
At the same time, the Kafka broker reported
```c
broker_1           | [2017-09-13 13:20:28,975] ERROR [KafkaApi-1] Number of alive brokers '1' does not meet the required replication factor '3' for the offsets topic (configured via 'offsets.topic.replication.factor'). This error can be ignored if the cluster is starting up and not all brokers are up yet. (kafka.server.KafkaApis)
```
Clearly, I can go back to an older Docker image, but this newer image is working with the JavaScript client, so I hope that Kaffe can use it too.

**Comment by objectuser** [comment]
@erikvullings What are the differences between the previous and new Docker image?

Also, were the brokers all up before you tried to connect? I wonder if the new image defaults to starting fewer brokers.

**Comment by davidsantoso** [comment]
@erikvullings I'm closing due to inactivity. Feel free to reopen if you'd still like to discuss this issue.

**Comment by erikvullings** [comment]
@davidsantoso Sorry for the late reply - I missed your response!
So the Docker-compose that I was using in the [link](https://gist.github.com/erikvullings/b6363444fb232477ef381b1bb7dcbb32) was based on the latest version of Kafka, which at that time didn't work, but when I fix it to version 3.2.1 (see below), it did work. Does that help.

`---
version: '2'
services:
  zookeeper:
    image: confluentinc/cp-zookeeper:3.2.1
    hostname: zookeeper
    extra_hosts:
    - "moby:127.0.0.1"
    ports:
      - "2181:2181"
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
      ZOOKEEPER_TICK_TIME: 2000

  broker:
    image: confluentinc/cp-kafka:3.2.1
    hostname: broker
    extra_hosts:
    - "moby:127.0.0.1"
    depends_on:
      - zookeeper
    ports:
      - '9092:9092'
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
      KAFKA_ADVERTISED_LISTENERS: 'PLAINTEXT://broker:9092'
      KAFKA_DEFAULT_REPLICATION_FACTOR: 1
      KAFKA_MESSAGE_MAX_BYTES: 100000000
      KAFKA_REPLICA_FETCH_MAX_BYTES: 100000000

  schema_registry:
    image: confluentinc/cp-schema-registry:3.2.1
    hostname: schema_registry
    # extra_hosts:
    # - "moby:127.0.0.1"
    depends_on:
      - zookeeper
      - broker
    ports:
      - '8081:8081'
    environment:
      SCHEMA_REGISTRY_HOST_NAME: schema_registry
      SCHEMA_REGISTRY_KAFKASTORE_CONNECTION_URL: 'zookeeper:2181'

  kafka-rest:
    image: confluentinc/cp-kafka-rest:3.2.1
    container_name: kafka-rest
    extra_hosts:
    - "moby:127.0.0.1"
    depends_on:
      - zookeeper
      - broker
    ports:
      - '8082:8082'
    environment:
      KAFKA_REST_ZOOKEEPER_CONNECT: 'zookeeper:2181'
      KAFKA_REST_LISTENERS: http://kafka-rest:8082
      KAFKA_REST_SCHEMA_REGISTRY_URL: http://schema-registry:8081
      KAFKA_REST_HOST_NAME: kafka-rest

  # kafka-manager:
  #   image: sheepkiller/kafka-manager
  #   depends_on:
  #   - zookeeper
  #   ports:
  #   - '9000:9000'
  #   links:
  #   - zookeeper
  #   environment:
  #   - ZK_HOSTS=zookeeper:2181
  #   - APPLICATION_SECRET=letmein
`

**Comment by uttamanand** [comment]
Were you able to find the solution for the above issue? 

I am getting an issue while producing and consuming messages. I have just one kafka broker and one zookeeper. So I have given replication factor of all topics as 1.
I don’t know from where is it trying to get replication factor as 3.

[2018-03-21   15:54:28,249] ERROR [KafkaApi-0] Number of alive brokers '1' does not meet   the required replication factor '3' for the offsets topic (configured via   'offsets.topic.replication.factor'). This error can be ignored if the cluster   is starting up and not all brokers are up yet. (kafka.server.KafkaApis)
